src/inputs/input/input.jsx

```diff
@@ -264,12 +377,26 @@ class Input extends React.PureComponent {
 
     onKeyDown(event) {
         const {
+            allowDecimals,
+            allowNegativeNumbers,
             onKeyDown,
         } = this.props;
 
+        const type = this.getType();
+
         if (isFunction(onKeyDown)) {
             onKeyDown(event);
         }
+
+        if (type === 'number') {
+            if (!allowDecimals && event.keyCode === DOT_KEY_CODE) {
+                event.preventDefault();
+            }
+
+            if (!allowNegativeNumbers && event.keyCode === MINUS_KEY_CODE) {
+                event.preventDefault();
+            }
+        }
```

Comment:
I really think we should consider disallowing (i.e. preventing default) _any_ characters other than 0-9 (and `.` and `-` depending on the values of `allowDecimals` and `allowNegativeNumbers`.

Thoughts?

Reply:
Maybe something like this?
```JavaScript
if (type === 'number') {
    const shouldAllowCharacter = (event.keyCode >= '0' && event.keyCode <= '9') || // character is a digit
        (allowDecimals && event.keyCode === DOT_KEY_CODE) || // character is the decimal separator / TODO/FIXME: some locales use comma instead of dot!
        (allowNegativeNumbers && event.keyCode === MINUS_KEY_CODE);
    
    if (!shouldAllowCharacter) {
        event.preventDefault();
    }
}
```

Reply:
Please check this commit https://github.com/saddlebackdev/react-cm-ui/pull/353/commits/8429b8ffd987325a6ae504db3aa2341b60de60ef

Reply:
Thank you, @groberts314 Let me look over different locales and link provided by you.

Reply:
Yeah, I repeat, discretionary/NAB. =)
Seriously!
Unless you want extra credit.

If you _REALLY_ do wanna do it:
1) Poor man's version: Allow comma as well as dot if `allowDecimals` is true.
2) Richer man's version:
    * Use one of the techniques from SO to detect the appropriate separator for the current local (should likely do this in constructor and assign it to a member field on `this`).
    * Change the condition in the new logic in `onKeyDown` for `allowDecimals` to allow that detected character